framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/pipeline/ExecutionStage.java

```diff
@@ -0,0 +1,31 @@
+/* Licensed under MIT 2023-2024. */
+package edu.kit.kastel.mcse.ardoco.core.pipeline;
+
+import java.util.List;
+import java.util.SortedMap;
+
+import edu.kit.kastel.mcse.ardoco.core.data.DataRepository;
+import edu.kit.kastel.mcse.ardoco.core.pipeline.agent.PipelineAgent;
+
+public abstract class ExecutionStage extends AbstractExecutionStage {
```

Comment:
Yes, I think the purpose of this class was to get rid of having a static "get" function in each stage where you create a new stage and initialize it with the given config (because such a function cannot be part of the abstract class). In hindsight, it makes more sense to just adjust the original abstract class (add additional configs as a constructor parameter)  instead and remove this class. Or use archrules to enforce the creation of a static get method in non-abstract classes extending abstract execution stage (for the sake of consistency)